pennylane/transforms/insert_ops.py

```diff
@@ -210,9 +210,15 @@ def f(w, x, y, z):
     """
     # decompose templates and their adjoints (which fixes a bug in the tutorial_error_mitigation demo)
     # TODO: change this to be cleaner and more robust
-    circuit = circuit.expand(
-        stop_at=lambda op: not hasattr(qml.templates, op.name) and not isinstance(op, Adjoint)
-    )
+    try:
+        circuit = circuit.expand(
+            stop_at=lambda op: not hasattr(qml.templates, op.name) and not isinstance(op, Adjoint)
+        )
+    except qml.QuantumFunctionError as e:
+        raise qml.QuantumFunctionError(
+            "The insert transform cannot transform a circuit with non-commuting observables. "
```

Comment:
is it worth getting into more technical detail about the execution pipeline here? In my opinion, no. Just wondering what reviewers might think.